examples/bridge/sn/src/dojo_token.cairo

```diff
@@ -0,0 +1,200 @@
+use starknet::{ContractAddress, ClassHash};
+use dojo::world::IWorldDispatcher;
+
+#[starknet::interface]
+trait IDojoToken<TState> {
+    // IWorldProvider
+    fn world(self: @TState,) -> IWorldDispatcher;
+
+    // IUpgradeable
+    fn upgrade(ref self: TState, new_class_hash: ClassHash);
+
+    // IERC20MetadataTotalSupply
+    fn total_supply(self: @TState,) -> u256;
+
+    // IERC20MetadataTotalSupplyCamel
+    fn totalSupply(self: @TState,) -> u256;
+
+    // IERC20Balance
+    fn balance_of(self: @TState, account: ContractAddress) -> u256;
+    fn transfer(ref self: TState, recipient: ContractAddress, amount: u256) -> bool;
+    fn transfer_from(
+        ref self: TState, sender: ContractAddress, recipient: ContractAddress, amount: u256
+    ) -> bool;
+
+    // IERC20BalanceCamel
+    fn balanceOf(self: @TState, account: ContractAddress) -> u256;
+    fn transferFrom(
+        ref self: TState, sender: ContractAddress, recipient: ContractAddress, amount: u256
+    ) -> bool;
+
+    // IERC20Allowance
+    fn allowance(self: @TState, owner: ContractAddress, spender: ContractAddress) -> u256;
+    fn approve(ref self: TState, spender: ContractAddress, amount: u256) -> bool;
+
+    // IERC20Metadata
+    fn decimals(self: @TState,) -> u8;
+    fn name(self: @TState,) -> felt252;
+    fn symbol(self: @TState,) -> felt252;
+
+    // IERC20SafeAllowance
+    fn decrease_allowance(
+        ref self: TState, spender: ContractAddress, subtracted_value: u256
+    ) -> bool;
+    fn increase_allowance(ref self: TState, spender: ContractAddress, added_value: u256) -> bool;
+
+    // IERC20SafeAllowanceCamel
+    fn decreaseAllowance(ref self: TState, spender: ContractAddress, subtractedValue: u256) -> bool;
+    fn increaseAllowance(ref self: TState, spender: ContractAddress, addedValue: u256) -> bool;
+
+    // IERC20Bridgeable
+    fn burn(ref self: TState, account: ContractAddress, amount: u256);
+    fn l2_bridge_address(self: @TState,) -> ContractAddress;
+    fn mint(ref self: TState, recipient: ContractAddress, amount: u256);
+
+    // WITHOUT INTERFACE !!!
```

Comment:
What do you mean by that?

Reply:
this is codegen it just say next fn are not included in another interface

Reply:
Indeed, if we could centralize codegen it could be awesome. With cainome, when plugin with stdin are supported, you have in fact a list of organized tokens instead of parsing the raw ABI that can be easily used to such tasks.

But I think that for now it's fine using this script to not delay this PR as tt's a specific use case which we have to see how it can be integrated in cainome for easy use.
Something like this from a rust code:
```rust
let trait_string = AbiParser::abi_to_trait("TraitName", manifest.abi);
```